sdkAngular/webpack.config.js

```diff
@@ -0,0 +1,222 @@
+const { resolve, join  } = require("path");
+
+const webpack = require("webpack");
+const nsWebpack = require("nativescript-dev-webpack");
+const nativescriptTarget = require("nativescript-dev-webpack/nativescript-target");
+const CopyWebpackPlugin = require("copy-webpack-plugin");
+const ExtractTextPlugin = require("extract-text-webpack-plugin");
+const { BundleAnalyzerPlugin } = require("webpack-bundle-analyzer");
+
+const { AotPlugin } = require("@ngtools/webpack");
+
+const mainSheet = `app.css`;
+
+module.exports = env => {
+    const platform = getPlatform(env);
+
+    // Default destination inside platforms/<platform>/...
+    const path = resolve(nsWebpack.getAppPath(platform));
+
+    const entry = {
+        // Discover entry module from package.json
+        bundle: `./${nsWebpack.getEntryModule()}`,
+
+        // Vendor entry with third-party libraries
+        vendor: `./vendor`,
+
+        // Entry for stylesheet with global application styles
+        [mainSheet]: `./${mainSheet}`,
+    };
+
+    const rules = getRules();
+    const plugins = getPlugins(platform, env);
+    const extensions = getExtensions(platform);
+
+    const config = {
+        context: resolve("./app"),
+        target: nativescriptTarget,
+        entry,
+        output: {
+            pathinfo: true,
+            path,
+            libraryTarget: "commonjs2",
+            filename: "[name].js",
+        },
+        resolve: {
+            extensions,
+
+            // Resolve {N} system modules from tns-core-modules
+            modules: [
+                "node_modules/tns-core-modules",
+                "node_modules",
+            ]
+        },
+        node: {
+            // Disable node shims that conflict with NativeScript
+            "http": false,
+            "timers": false,
+            "setImmediate": false,
+            "fs": "empty",
+        },
+        module: { rules },
+        plugins,
+    };
+
+    if (env.snapshot) {
+        plugins.push(new nsWebpack.NativeScriptSnapshotPlugin({
+            chunk: "vendor",
+            projectRoot: __dirname,
+            webpackConfig: config,
+            targetArchs: ["arm", "arm64", "ia32"],
+            tnsJavaClassesOptions: { packages: ["tns-core-modules" ], modules: [join(__dirname, "app/main-activity.android.js")] },
```

Comment:
I suggest to extract the modules property to a const and explain why it's needed.